roslibrust/src/ros1/tcpros.rs

```diff
@@ -0,0 +1,117 @@
+use byteorder::{LittleEndian, ReadBytesExt, WriteBytesExt};
+use std::io::{Cursor, Read, Write};
+
+pub struct ConnectionHeader {
+    pub caller_id: String,
+    pub latching: bool,
+    pub msg_definition: String,
+    pub md5sum: String,
+    pub topic: String,
+    pub topic_type: String,
+    pub tcp_nodelay: bool,
+}
+
+impl ConnectionHeader {
+    pub fn from_bytes(header_data: &[u8]) -> std::io::Result<ConnectionHeader> {
+        let mut cursor = Cursor::new(header_data);
+        let header_length = cursor.read_u32::<LittleEndian>()?;
+        if header_length as usize > header_data.len() {
+            return Err(std::io::ErrorKind::InvalidInput.into());
+        }
+
+        let mut msg_definition = String::new();
+        let mut caller_id = String::new();
+        let mut latching = false;
+        let mut md5sum = String::new();
+        let mut topic = String::new();
+        let mut topic_type = String::new();
+        let mut tcp_nodelay = false;
+
+        // TODO: Unhandled: error, persistent
+
```

Comment:
Prolly good to stick this link in here for future us:
http://wiki.ros.org/ROS/Connection%20Header